ipatests/test_integration/test_adtrust_install.py

```diff
@@ -0,0 +1,190 @@
+             '--add-sids', '-U'])
+        res = self.master.run_command(['testparm', '-s'])
+        assert 'ERROR' not in (res.stdout_text + res.stderr_text)
+
+    def test_add_agent_not_allowed(self):
+        """Check that add-agents can be run only by Admins
+        """
+        user = "nonadmin"
+        passwd = "Secret123"
+        tasks.create_active_user(self.master, user, passwd,
+                                 first=user, last=user)
+        tasks.kinit_as_user(self.master, user, passwd)
+        host = self.replicas[0].hostname
+        data_fmt = '{{"method":"trust_enable_agent","params":[["{}"],{{}}]}}'
+        cmd_args = [
+            paths.BIN_CURL,
+            '-H', 'referer:https://{}/ipa'.format(host),
+            '-H', 'Content-Type:application/json',
+            '-H', 'Accept:applicaton/json',
+            '--negotiate', '-u', ':',
+            '--cacert', paths.IPA_CA_CRT,
+            '-d', data_fmt.format(host),
+            '-X', 'POST', 'https://{}/ipa/json'.format(host)]
+        res = self.master.run_command(cmd_args)
+        tasks.kinit_admin(self.master)
+        expected_re = 'Insufficient access: not allowed to remotely add agent'
+        assert re.search(expected_re, res.stdout_text)
+
+    def test_add_agent_on_stopped_replica(self):
+        """ Check ipa-adtrust-install --add-agents when the replica is stopped.
+
+        Scenario: stop a replica
+        Call ipa-adtrust-install --add-agents and configure the stopped replica
+        as a new agent.
+        The tool must detect that the replica is stopped and warn that
+        a part of the configuration failed.
+
+        Test for https://pagure.io/freeipa/issue/8148
+        """
+        self.unconfigure_replica_as_agent(self.replicas[0])
+        self.replicas[0].run_command(['ipactl', 'stop'])
+
+        cmd_input = (
+            # admin password:
+            self.master.config.admin_password + '\n' +
+            # WARNING: The smb.conf already exists. Running ipa-adtrust-install
+            # will break your existing samba configuration.
+            # Do you wish to continue? [no]:
+            'yes\n'
+            # Enable trusted domains support in slapi-nis? [no]:
+            '\n' +
+            # WARNING: 1 IPA masters are not yet able to serve information
+            # about users from trusted forests.
+            # Installer can add them to the list of IPA masters allowed to
+            # access information about trusts.
+            # If you choose to do so, you also need to restart LDAP service on
+            # those masters.
+            # Refer to ipa-adtrust-install(1) man page for details.
+            # IPA master[replica1.testrelm.test]?[no]:
+            'yes\n'
+        )
+        res = self.master.run_command(['ipa-adtrust-install', '--add-agents'],
+                                      stdin_text=cmd_input)
+        self.replicas[0].run_command(['ipactl', 'start'])
```

Comment:
if `ipa-adtrust-install` fails then replica will not be started and next tests will fail. It is better to place it in the `finally` block

Reply:
OK, updated.